src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -50,17 +50,17 @@
 import com.puppycrawl.tools.checkstyle.api.FileText;
 import com.puppycrawl.tools.checkstyle.api.Filter;
 import com.puppycrawl.tools.checkstyle.api.FilterSet;
-import com.puppycrawl.tools.checkstyle.api.MessageDispatcher;
 import com.puppycrawl.tools.checkstyle.api.RootModule;
 import com.puppycrawl.tools.checkstyle.api.SeverityLevel;
 import com.puppycrawl.tools.checkstyle.api.SeverityLevelCounter;
 import com.puppycrawl.tools.checkstyle.api.Violation;
+import com.puppycrawl.tools.checkstyle.api.ViolationDispatcher;
```

Comment:
Considering this interface seats in the api package, should we go via deprication process first, so we introduce a new interface but still keep the old one marked as deprecated? Otherwise, this PR may contain breaking changes. @romani what are your thoughts on this?

Reply:
Yes, we need to think on deprecation first. To let all plugins to migrate first, and if no known tools left, do deletion.
Is it possible?